perfkitbenchmarker/providers/ibmcloud/ibmcloud.py

```diff
@@ -0,0 +1,989 @@
+    if not self._token or self._force or self._token_time < (time.time() - 3000):
+      self.SetToken()
+
+  @vm_util.Retry(max_retries=3, timeout=vm_util.DEFAULT_TIMEOUT)
+  def GetToken(self):
+    """Get a user token."""
+    token = None
+    _req_data = {
+      'grant_type': 'urn:ibm:params:oauth:grant-type:apikey',
+      'response_type': 'cloud_iam',
+      'apikey': self._apikey
+    }
+    count = 1
+    while token is None:
+      if not self._silent:
+        logging.info('Sending a POST request to get a token: %s', self._endpoint)
+      resp = requests.request('POST', self._endpoint, data=_req_data, headers=None,
+                              timeout=(HTTP_TIMEOUT_CONNECT, HTTP_TIMEOUT))
+      if resp is not None:
+        if self._verbose:
+          logging.info('response=%s', str(resp))
+          if resp.text:
+            logging.info('response text=%s', resp.text)
+        if resp.status_code >= 200 and resp.status_code <= 299:
+          resp = json.loads(resp.text)
+          if 'access_token' in resp:
+            token = resp['access_token']
+            if self._verbose:
+              logging.info('token: %s', token)
+            break
+        else:
+          logging.error('POST: %s, response status_code: %s, resp: %s',
+                        self._endpoint, resp.status_code, str(resp))
+      else:
+        logging.info('Request POST: %s, no response returned', self._endpoint)
+      count += 1
+      if count > 3:
+        break
+      time.sleep(5)
+    return token
+
+  def Token(self):
+    """Returns the token."""
+    return self._token
+
+  def SetToken(self):
+    """Gets a new token."""
+    try:
+      self._token = self.GetToken()
+      self._token_time = time.time()
+    except Exception as err:
+      raise IbmCloudError('Authorization failed: %s' % err)
+
+  def Request(self, method, path, data=None, headers=None, timeout=None, session=None):
+    """Constructs base rest api calls to run against IBM Cloud regional api server.
+    Args:
+      method: one of GET, POST, PUT, PATCH or DELETE.
+      path: rest api uri.
+      data: body data.
+      headers: header data if any.
+      timeout: timeout for the request
+      session: session object if available.
+    """
+    path = '/' + self._version + '/%s' % (path)
+    if 'limit' in path:
+      path += '&' + self._generation
+    else:
+      path += '?' + self._generation
+
+    h = {'Authorization': 'Bearer ' + self._token}
+    if self._trace:
+      h['trace'] = 'on'
+    if data is not None and not isinstance(data, IOBase):
+      h['Content-Type'] = 'application/json'
+      data = json.dumps(data)
+    if headers:
+      h.update(headers)
+    if self._verbose:
+      h['trace'] = 'on'
+      curl = '>>> curl -X %s' % method
+      for k, v in h.items():
+        curl += ' -H "%s: %s"' % (k, v)
+      if data:
+        curl += ' -d ' + '"' + str(data).replace('"', '\\"') + '"'
```

Comment:
Is there an IBM cloud SDK like `gcloud` or `aws` or `az` commands for GCP, AWS, Azure?

Reply:
There's ibmcloud CLI and REST API's, no SDK.